qiskit_experiments/library/randomized_benchmarking/rb_analysis.py

```diff
@@ -235,4 +233,40 @@ def _extra_database_entry(self, fit_data: curve.FitData) -> List[AnalysisResultD
                                 device_components=[Qubit(i) for i in qubits],
                             )
                         )
+
         return extra_entries
+
+    def _run_analysis(
+        self, experiment_data: ExperimentData
+    ) -> Tuple[List[AnalysisResultData], List["pyplot.Figure"]]:
+
+        if self.options.gate_error_ratio is None:
+            gate_error_ratio = experiment_data.metadata.get("gate_error_ratio", None)
+
+            if gate_error_ratio is None:
+                # For backward compatibility when loading old experiment data.
```

Comment:
Should we raise the deprecation warning here so its clearer in the code that this is a deprecated path?

Reply:
Good question. As I wrote in the comment, calling `RBUtils.get_error_dict_from_backend` raises deprecation warning.  I agree writing explicit warning here helps maintainer, but this means user will see the warning twice.